prometheus considerations + link KSM values

jessegoodier opened the pull request:
## What does this PR change?

Add link to documentation around disabling Prometheus.
Add section for disabling KSM

## Does this PR rely on any other PRs?

- NA



## How does this PR impact users? (This is the kind of thing that goes in release notes!)

- Reduces the support issues from users implementing kubecost on existing Prometheus
- Multiple KSM pods are inefficient

## Links to Issues or ZD tickets this PR addresses or fixes

NA

## How was this PR tested?

```
helm template kubecost \
    --repo https://kubecost.github.io/cost-analyzer/ cost-analyzer \
    --namespace kubecost --create-namespace \
    --set prometheus.nodeExporter.enabled=false \
    --set prometheus.serviceAccounts.nodeExporter.create=false \
    --set prometheus.kubeStateMetrics.enabled=false|ag 'statemetric|state-metric'
```
![image](https://user-images.githubusercontent.com/31039225/186249653-cb41d471-9671-46a5-a45e-178e28aa6327.png)


## Have you made an update to documentation?

yes: https://github.com/kubecost/docs/blob/main/custom-prom.md

linhlam-kc reviewed (approved):
LGTM

srpomeroy commented:
The root readme.md and the cost-analyzer/readme.md both reference `prometheus.kube-state-metrics.disabled`. We should consider either updating the readmes or switching to that value if it makes more sense.

jessegoodier commented:
I'm not merging until Bolt gives his ok. 

It is odd. Prometheus has updated to `condition: kubeStateMetrics.enabled`
(not that this matters): https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/Chart.yaml#L29)

From what we have: 
`  - condition: kube-state-metrics.disabled`


but disabled doesn't work:
```
helm template kubecost \
    --repo https://kubecost.github.io/cost-analyzer/ cost-analyzer \
    --namespace kubecost --create-namespace \
    --set prometheus.nodeExporter.enabled=false \
    --set prometheus.serviceAccounts.nodeExporter.create=false \
    --set prometheus.kubeStateMetrics.disabled=true \
    | grep -iE 'kube-state-metrics/templates/deployment' -A5
```
output:
```# Source: cost-analyzer/charts/prometheus/charts/kube-state-metrics/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: kubecost-kube-state-metrics
...
```

whereas enabled=false works as captured in initial testing section does not deploy KSM.

srpomeroy commented:
For `disabled` it should be `--set prometheus.kube-state-metrics.disabled=true` note the hyphens.
https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/charts/prometheus/charts/kube-state-metrics/templates/deployment.yaml#L1

jessegoodier commented:
> For `disabled` it should be `--set prometheus.kube-state-metrics.disabled=true` note the hyphens. https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/charts/prometheus/charts/kube-state-metrics/templates/deployment.yaml#L1

sure enough:
```
helm template kubecost \
    --repo https://kubecost.github.io/cost-analyzer/ cost-analyzer \
    --namespace kubecost --create-namespace \
    --set prometheus.kube-state-metrics.disabled=true 
```
also disables it.

So, which is better @mbolt35 ?
--set prometheus.kubeStateMetrics.enabled=false
or
--set prometheus.kube-state-metrics.disabled=true 

or just do both?

mbolt35 commented:
@jessegoodier I'll look at this today at some point. A few things to note before I dive into the specifics: we don't require the KSM deployment anymore, as our cost-model pod emits all of the KSM metrics we use. (Although deploying KSM is fine because we handle duplicate metrics cases anyways).

jessegoodier commented:
maybe we should just set KSM to disabled here by default?

jessegoodier commented:
updated per @mbolt35's comment on KSM on other PR regarding both KSM enable/disable. 
also added comment for additional details.

I believe this is good to go. Please reapprove if you agree.

mbolt35 reviewed (approved):
Love the added troubleshoot links! Also, yes, I think at some point we will probably want to unify the whole `kubeStateMetrics` and `kube-state-metrics` duality, but the comment and explicit defaults capture it well. Thanks Jesse!
